appcelerator/titanium_mobile
# refactor(android): remove "WebViewClient.jar" from SDK

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28241

**Summary:**
- This JAR was used to handle `WebView` client certificate requests on Android 4.x.
- Now that Titanium's min OS version has been raised to Android 5.0, JAR is no longer needed.
- Replaced `AsyncTask` with `Thread` since we were only using task's `doInBackground()` method.

**Test:**
1. Build and run the below on Android.
2. Tap on the "Download Certificate" button.
3. Tap on the `badssl.com-client.p12` link.
4. If Android asks for a password then enter: `badssl.com`
5. Go to Android's system `Settings -> Security -> Advanced -> Encryption & Credentials` screen.
6. Tap on `Install Certificate -> VPN & app user certificate` screen.
7. Select downloaded certificate from "Downloads" folder and install it.
8. Note that you might need to enable a passcode/PIN on the device.
9. Go back to the Titanium app.
10. Tap on the "Show WebView" button.
11. Verify an alert dialog is shown asking for a certificate.
12. Select the installed certificate.
13. Verify that a green webpage is shown. (A red webpage is shown if client certificate is NOT selected.)

```javascript
const parentWindow = Ti.UI.createWindow();
const downloadButton = Ti.UI.createButton({
	title: "Download Certificate",
	top: "30%",
});
downloadButton.addEventListener("click", () => {
	Ti.Platform.openURL("https://badssl.com/download/");
});
parentWindow.add(downloadButton);
const webButton = Ti.UI.createButton({
	title: "Show WebView",
	bottom: "30%",
});
webButton.addEventListener("click", () => {
	const webWindow = Ti.UI.createWindow();
	webWindow.add(Ti.UI.createWebView({
		url: "https://client.badssl.com/",
		width: Ti.UI.FILL,
		height: Ti.UI.FILL,
	}));
	webWindow.open();
});
parentWindow.add(webButton);
parentWindow.open();
```
